ndk/src/media/mod.rs

```diff
@@ -3,20 +3,11 @@
 //! See also [the NDK docs](https://developer.android.com/ndk/reference/group/media)
 #![cfg(feature = "media")]
 
-mod error;
 pub mod image_reader;
 pub mod media_codec;
 
-pub use error::NdkMediaError;
-use std::{mem::MaybeUninit, ptr::NonNull};
-
-pub type Result<T, E = NdkMediaError> = std::result::Result<T, E>;
-
-fn construct<T>(with_ptr: impl FnOnce(*mut T) -> ffi::media_status_t) -> Result<T> {
-    let mut result = MaybeUninit::uninit();
-    let status = with_ptr(result.as_mut_ptr());
-    NdkMediaError::from_status(status).map(|()| unsafe { result.assume_init() })
-}
+use crate::media_error::{construct, NdkMediaError, Result};
```

Comment:
As discussed previously, in order to not make this a breaking change (which should otherwise be clearly marked in the `README`) `NdkMediaError` and `Result` should be `pub use`. In fact, as a bugfix `MediaErrorResult` should be made publicly accessible too.

I'd like to get that in as a preliminary PR to separate concerns from the `AMidi` feature, and also revise how the errors are structured currently. Mind if I address that first?

Reply:
No, please go ahead. I'll find other potential issues in the current pull request by running some tests. Thanks for explaining the current situation!

Reply:
This separation is now open at https://github.com/rust-mobile/ndk/pull/399